test/lib/net/dhcp_test.rb

```diff
@@ -56,6 +56,15 @@ class DhcpTest < ActiveSupport::TestCase
     assert_equal record2, record1
   end
 
+  test "record should be equal if one record has no filename" do
+    record1 = Net::DHCP::Record.new(:mac => "aa:bb:cc:dd:ee:ff",
+                                    :network => "127.0.0.0", :ip => "127.0.0.1", "proxy" => smart_proxies(:one))
+    record2 = Net::DHCP::Record.new(:filename => "pxelinux.0", :mac => "aa:bb:cc:dd:ee:ff",
+                                    :network => "127.0.0.0", :ip => "127.0.0.1", "proxy" => smart_proxies(:one))
+    assert_equal record1, record2
+    assert_equal record2, record1
```

Comment:
We are exercising commutative property of `==` operation. You can have bugs in the method (it's `this` vs `other` and vice versa). Usually nil exceptions are thrown if not tested properly.